apps/duellen/app/components/Intro.jsx

```diff
@@ -66,14 +66,46 @@ export default class Intro extends React.Component{
       action: 'Started Quiz'
     });
   }
+  getweek(){
+    Date.prototype.getWeek = function (dowOffset) {      
+          dowOffset = typeof(dowOffset) == 'int' ? dowOffset : 0; //default dowOffset to zero
+          var newYear = new Date(this.getFullYear(),0,1);
+          var day = newYear.getDay() - dowOffset; //the day of week the year begins on
+          day = (day >= 0 ? day : day + 7);
+          var daynum = Math.floor((this.getTime() - newYear.getTime() - 
+          (this.getTimezoneOffset()-newYear.getTimezoneOffset())*60000)/86400000) + 1;
+          var weeknum;
+          //if the year starts before the middle of a week
+          if(day < 4) {
+              weeknum = Math.floor((daynum+day-1)/7) + 1;
+              if(weeknum > 52) {
+                  nYear = new Date(this.getFullYear() + 1,0,1);
+                  nday = nYear.getDay() - dowOffset;
+                  nday = nday >= 0 ? nday : nday + 7;
+                  /*if the next year starts before the middle of
+                    the week, it is week #1 of that year*/
+                  weeknum = nday < 4 ? 1 : 53;
+              }
+          }
+          else {
+              weeknum = Math.floor((daynum+day-1)/7);
+          }
+          return weeknum;
+      };
+      var date = this.state.quizData.publication_date
+      var mydate = new Date(Number(date.slice(0,4)), Number(date.slice(5,7))-1,Number(date.slice(8,10)));
+      this.setState({
+        week: mydate.getWeek(),
+      });     
+  }
```

Comment:
The same `getWeek` appears both in `Start.jsx` and `Intro.jsx`. I'd recommend to place it in a separate file and importing it in both places. Or maybe there is an npm package for this.